server/oauth2_test.go

```diff
@@ -27,8 +27,13 @@ func TestParseAuthorizationRequest(t *testing.T) {
 
 		queryParams map[string]string
 
-		wantErr    bool
-		exactError *authErr
```

Comment:
With fields for different error types test looks a little bit complicated now. You need to know a lot about auth error structure to add new test cases. Can we leave the `exactError` here?